### PR TITLE
Enhance shutdown service to handle pre-terminated services.

### PR DESCRIPTION
**Pull request by YoEight**
Fixed: Enhance shutdown service to handle pre-terminated services.

Updates the shutdown service to handle cases where services have already terminated before the shutdown process begins. The change ensures that the shutdown sequence gracefully bypasses pre-terminated services, preventing potential errors and improving reliability during system shutdown.

### CODE REVIEW

**Review by timothycoleman — APPROVED** [review]

File: src/EventStore.Core/Services/ShutdownService.cs [hunk 1]
```diff
@@ -6,72 +6,116 @@
...
-				correlationId: Guid.NewGuid(),
-				exitProcess: _exitProcess,
-				shutdownHttp: _shutdownHttp));
+			ShutDownCore();
 		}
 	}
+
+	public void Handle(SystemMessage.PeripheralShutdownTimeout message) {
+		if (_state is not State.ShuttingDownPeriphery)
+			return;
```
Comment (RagingKore) [inline src/EventStore.Core/Services/ShutdownService.cs:106]: We should log DEBUG on this exit too

**Review by RagingKore — CHANGES_REQUESTED** [review]

File: src/EventStore.Core/Services/ShutdownService.cs [hunk 2]
```diff
@@ -6,72 +6,116 @@
...
+
+	public ShutdownService(IPublisher mainQueue, VNodeInfo nodeInfo, TimeSpan? shutdownTimeout = null) {
 		_mainQueue = mainQueue;
 		_nodeInfo = nodeInfo;
+		_shutdownTimeout = shutdownTimeout ?? PeripheryShutdownTimeout;
 	}
 
 	public void Handle(SystemMessage.RegisterForGracefulTermination message) {
+		if (_state is not State.Running) {
+			Log.Warning("Component {ComponentName} registered for graceful shutdown while the server is shutting down",
```
[the remark this answers is not in the document]
Comment (timothycoleman) [reply src/EventStore.Core/Services/ShutdownService.cs:48]: I’m not sure I follow this

[the remark this answers is not in the document]
Comment (RagingKore) [reply src/EventStore.Core/Services/ShutdownService.cs]: The component never got registered. The log message should reflect that.

**Review by timothycoleman — APPROVED** [review]

**Review by RagingKore — APPROVED** [review]
LGTM

**Pull request merged**